[MOD] Features requested for kick/ban.

Ianardo-DiCaprio opened the pull request:
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes
Added features to kick and ban requested in #2649. This will now send an embed to the user when they are kicked/banned from a server detailing if it was a kick or ban, from what server, <s>the moderator that issued the kick/ban</s> and the reason for the kick/ban. Also blacked.

Ianardo-DiCaprio committed:
New Features For Warning

Ianardo-DiCaprio committed:
blacked

Ianardo-DiCaprio committed:
Accidentally added my changed warnings.py to wrong branch.

Ianardo-DiCaprio committed:
Add files via upload

Ianardo-DiCaprio committed:
-l 99

Ianardo-DiCaprio committed:
-l 99

yamikaitou commented:
Please allow this feature to be optional (default off)

Ianardo-DiCaprio committed:
Added a `[p]modset toggledm` command.

Added a `[p]modset toggledm` command to toggle whether or not a message is sent to the user on kick/ban.
Also added the this toggle to show as yes or no when using [p]modset.

Ianardo-DiCaprio committed:
revert warnings.py changes

Ianardo-DiCaprio committed:
Add files via upload

Ianardo-DiCaprio committed:
fix

Ianardo-DiCaprio committed:
Say that 7 days is max

Ianardo-DiCaprio committed:
Review changes

Ianardo-DiCaprio committed:
docstring change

Ianardo-DiCaprio committed:
black -l 99

NeuroAssassin commented on redbot/cogs/mod/kickban.py:
```diff
@@ -179,6 +179,20 @@ class KickBanMixin(MixinMeta):
             return
         audit_reason = get_audit_reason(author, reason)
         try:
+            if reason == None:
```
Comparison with `None`s should be using `is`, not `==`.

NeuroAssassin commented on redbot/cogs/mod/kickban.py:
```diff
@@ -179,6 +179,20 @@ class KickBanMixin(MixinMeta):
+            if reason == None:
+                reason = "No reason was given."
+            else:
+                reason = reason
```
You don't need the `else` here for redefining the reason to itself.

NeuroAssassin commented on redbot/cogs/mod/kickban.py:
```diff
@@ -179,6 +179,20 @@ class KickBanMixin(MixinMeta):
+                reason = "No reason was given."
+            else:
+                reason = reason
+            self == self.bot.get_cog("Mod")
```
Is this supposed to be a `=`?  And if so, `self` is already the Mod cog, so this wouldn't be needed.

NeuroAssassin commented on redbot/cogs/mod/kickban.py:
```diff
@@ -218,7 +232,22 @@ class KickBanMixin(MixinMeta):
-
+        author = ctx.author
+        guild = ctx.guild
+        if reason == None:
```
Should use `is` instead of `==` when comparing with `None`.

NeuroAssassin commented on redbot/cogs/mod/kickban.py:
```diff
@@ -218,7 +232,22 @@ class KickBanMixin(MixinMeta):
+        if reason == None:
+            reason = "No reason was given."
+        else:
+            reason = reason
```
You don't need the `else` here for redefining `reason`.

Ianardo-DiCaprio committed:
Review changes.

NeuroAssassin commented on redbot/cogs/mod/kickban.py:
```diff
@@ -218,7 +232,22 @@ class KickBanMixin(MixinMeta):
+            reason = "No reason was given."
+        else:
+            reason = reason
+        self == self.bot.get_cog("Mod")
```
Same thing as above, if its supposed to be a `=`, then it isn't needed as `self` is already the Mod cog.

NeuroAssassin commented on redbot/cogs/mod/kickban.py:
```diff
@@ -179,6 +179,20 @@ class KickBanMixin(MixinMeta):
+                reason = reason
+            self == self.bot.get_cog("Mod")
+            toggle = await self.settings.guild(guild).toggle_dm()
+            if toggle == True:
```
This can be changed to just `if toggle:`.

NeuroAssassin commented on redbot/cogs/mod/kickban.py:
```diff
@@ -218,7 +232,22 @@ class KickBanMixin(MixinMeta):
+            reason = reason
+        self == self.bot.get_cog("Mod")
+        toggle = await self.settings.guild(guild).toggle_dm()
+        if toggle == True:
```
Can be changed to just `if toggle:`.

NeuroAssassin reviewed (changes requested):
I've made a list of things could use a few improvements/changes.  Here's the list:

Ianardo-DiCaprio committed:
Review changes

Ianardo-DiCaprio committed:
Add files via upload

mikeshardmind commented on redbot/cogs/mod/kickban.py:
```diff
@@ -218,7 +232,22 @@ class KickBanMixin(MixinMeta):
+                em = discord.Embed(
+                    title=_("**You have been banned from {guild}.**").format(guild=guild)
+                )
+                em.add_field(name=_("**Moderator**"), value=author.name, inline=False)
```
Showing the reason here is fine. Showing the moderator is not unless there is another setting for this.

Ianardo-DiCaprio replied:
Would you prefer no moderator or an extra setting for it?

mikeshardmind commented on redbot/cogs/mod/settings.py:
```diff
@@ -199,3 +203,20 @@ class ModSettings(MixinMeta):
+
+    @modset.command()
+    @commands.guild_only()
+    async def toggledm(self, ctx: commands.Context):
```
For consistency with other core commands with configuration, this should accept a bool for setting it directly. If you'd like it to act as a toggle when that bool is not provided, that is fine.

mikeshardmind reviewed (changes requested):
In addition to some feedback another contributor pointed out, there are a few issues of design here.

Ianardo-DiCaprio committed:
Review changes.

Ianardo-DiCaprio committed:
Sinbad changes

Ianardo-DiCaprio committed:
black

Ianardo-DiCaprio committed:
Change to days.

Ianardo-DiCaprio committed:
Review changes + fix

Made `[p]warningset warnchannel` set the channel to None again if no channel is said.
Made "("{user} has been warned."" send if it is being sent in channels in the channel the command is used or the set channel if one is set.
Removed the message saying that the user has been warned in their DM's
Added ctx.tick() to the warn command when the command has completed instead.
Fixed sending the warn in the channel if the channel is set.

Ianardo-DiCaprio committed:
black -l 99

Ianardo-DiCaprio committed:
Removed unneeded import

Ianardo-DiCaprio committed:
Review change

Ianardo-DiCaprio committed:
Why did it unblack itself from removing 1 thing...

Ianardo-DiCaprio committed:
_ translator

Ianardo-DiCaprio committed:
black annoys me

Ianardo-DiCaprio committed:
Merge pull request #1 from Cog-Creators/V3/develop

update

Ianardo-DiCaprio committed:
Merge pull request #2 from Cog-Creators/V3/develop

update

Ianardo-DiCaprio committed:
Update settings.py

Ianardo-DiCaprio committed:
Update kickban.py

Ianardo-DiCaprio committed:
Update settings.py

Ianardo-DiCaprio committed:
Update settings.py

Ianardo-DiCaprio committed:
Update settings.py

Ianardo-DiCaprio committed:
Fix not valid value error.

Ianardo-DiCaprio committed:
Added changelog

Ianardo-DiCaprio committed:
Changelog

Ianardo-DiCaprio committed:
Update redbot/cogs/mod/kickban.py

Co-Authored-By: DevilXD <DevilXD@users.noreply.github.com>

Ianardo-DiCaprio committed:
Merge pull request #3 from Cog-Creators/V3/develop

update

Ianardo-DiCaprio committed:
Added Changelog

Ianardo-DiCaprio committed:
Review change

Ianardo-DiCaprio committed:
Update mod.py

Ianardo-DiCaprio committed:
Removed registration_task.cancel

Removed registration_task.cancel as registration_task was removed in #2901

Ianardo-DiCaprio committed:
Reset warnings

Ianardo-DiCaprio committed:
Reset mod changes

Ianardo-DiCaprio committed:
Delete 2814.enhance.rst

Ianardo-DiCaprio committed:
Remove self.registration_task.cancel()

Removed self.registration_task.cancel() as registration_task was removed in a previous PR and this now causes an error when you unload/reload mod.

Ianardo-DiCaprio committed:
Added changelog

Ianardo-DiCaprio committed:
Merge pull request #6 from Ianardo-DiCaprio/patch-4

Patch 4

Ianardo-DiCaprio committed:
Merge pull request #5 from Ianardo-DiCaprio/patch-3

Patch 3

Ianardo-DiCaprio committed:
Merge branch 'all' into patch-2

Ianardo-DiCaprio committed:
Update settings.py

Ianardo-DiCaprio committed:
Update settings.py

Ianardo-DiCaprio committed:
Update mod.py

Ianardo-DiCaprio committed:
Update kickban.py

Ianardo-DiCaprio committed:
Delete 2816.enhance.rst

Ianardo-DiCaprio committed:
Delete 2922.bugfix.rst

Ianardo-DiCaprio committed:
black and fix some weird thing github did.

Ianardo-DiCaprio committed:
Add missing await.

Ianardo-DiCaprio committed:
Fix found bug

Added an if to make sure the bots role is higher than the users and that the user isn't guild owner before sending message to user.

Was previously sending the message even if the bot failed to ban due to hierarchy.

Ianardo-DiCaprio committed:
Added another check

Don't send embed to user if  author != user

Ianardo-DiCaprio committed:
I'm actually just retarded.

Ianardo-DiCaprio committed:
Add same check to kick.

Ianardo-DiCaprio committed:
black
